Source/InstrumentEditorVRC7.cpp

```diff
@@ -30,9 +30,52 @@
 #include "InstrumentEditorVRC7.h"
 #include "Clipboard.h"
 
-static unsigned char default_inst[(16+3)*16] = 
+#include "FamiTracker.h"
+#include "Settings.h"
+
+// based off NSFPlay emu2413's hardware patch scheme
+#define OPLL_TONE_NUM 9
+static unsigned char default_inst[OPLL_TONE_NUM][(16 + 3) * 8] =
 {
-#include "APU/nsfplay/xgm/devices/Sound/legacy/vrc7tone_nuke.h" 
+	{
+#include "APU/nsfplay/xgm/devices/Sound/legacy/vrc7tone_nuke.h"
+	},
+	{
+#include "APU/nsfplay/xgm/devices/Sound/legacy/vrc7tone_rw.h"
+	},
+	{
+#include "APU/nsfplay/xgm/devices/Sound/legacy/vrc7tone_ft36.h"
+	},
+	{
+#include "APU/nsfplay/xgm/devices/Sound/legacy/vrc7tone_ft35.h"
+	},
+	{
+#include "APU/nsfplay/xgm/devices/Sound/legacy/vrc7tone_mo.h"
+	},
+	{
+#include "APU/nsfplay/xgm/devices/Sound/legacy/vrc7tone_kt2.h"
+	},
+	{
+#include "APU/nsfplay/xgm/devices/Sound/legacy/vrc7tone_kt1.h"
+	},
+	{
+#include "APU/nsfplay/xgm/devices/Sound/legacy/2413tone.h"
+	},
+	{
+#include "APU/nsfplay/xgm/devices/Sound/legacy/281btone.h"
+	},
+};
+
+enum PatchTone {
```

Comment:
do you feel confident you won't desync the order/items of default_inst and PatchTone?

Reply:
The order/items of the patches in default_inst and PatchTone are based on the configuration settings, which are based on NSFPlay's number and order of hardware patch banks inside its own implementation of VRC7.